packages/react-tooltip/src/components/TooltipProvider/useTooltipManager.ts

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { ShowTooltipArgs, TooltipManager, TooltipTriggerReason } from '@fluentui/react-tooltip-trigger';
+
+const useTimeout = () => {
+  type TimeoutState = {
+    id: number | undefined;
+    set: (fn: () => void, delay: number) => void;
+    clear: () => void;
+  };
+
+  const state = React.useRef<TimeoutState>({
+    id: undefined,
+    set: (fn: () => void, delay: number) => {
+      state.clear();
+      state.id = window.setTimeout(fn, delay);
```

Comment:
if you use the `window` variable you might want a guard since it is undefined in SSR

Reply:
I'll guard it with `?.`
However, in practice this should never be invoked during SSR, since it happens in response to user interaction.

Reply:
In practice I've never actually seen `window` used with `setTimeout` even though it's correct.

`setTimeout` might be better since AFAIK in SSR it's actually mocked to be a noop